File: src/main/java/wooyoung/tom/simplespringboot/repository/SpringDataJpaMemberRepository.java
package wooyoung.tom.simplespringboot.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import wooyoung.tom.simplespringboot.domain.Member;

import java.util.Optional;

public interface SpringDataJpaMemberRepository extends JpaRepository<Member, Long>, MemberRepository {

    @Override
    Optional<Member> findByName(String name);
}

File: src/main/java/wooyoung/tom/simplespringboot/repository/member/SpringDataJpaMemberRepository.kt
package wooyoung.tom.simplespringboot.repository.member

import org.springframework.data.jpa.repository.JpaRepository
import wooyoung.tom.simplespringboot.domain.Member
import java.util.*

interface SpringDataJpaMemberRepository: JpaRepository<Member, Long>, MemberRepository {

    override fun findByName(name: String): Optional<Member>
}